File: app/src/main/java/com/constraint/vagabond/main/search/adapter/AreaHolder.java
package com.constraint.vagabond.main.search.adapter;

import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.constraint.vagabond.R;
import com.squareup.picasso.Picasso;

import androidx.recyclerview.widget.RecyclerView;

public class AreaHolder extends RecyclerView.ViewHolder implements AreaHolderView {

  private final TextView areaName;
  private final Button moreInfoBtn;
  private final ImageView backgroundImage;

  AreaHolder(final View view) {
    super(view);
    areaName = view.findViewById(R.id.area_name);
    moreInfoBtn = view.findViewById(R.id.more_info);
    backgroundImage = view.findViewById(R.id.card_background);
  }

  @Override
  public void setAreaTitle(String areaTitle) {
    areaName.setText(areaTitle);
  }

  @Override
  public void setAreaImage(String imageUrl) {
    Picasso.get()
        .load(imageUrl)
        .placeholder(R.drawable.baseline_report_problem_24)
        .fit()
        .centerCrop()
        .into(backgroundImage);
  }

  @Override
  public void setMoreInfoButton() {}
}

File: app/src/main/java/com/constraint/vagabond/main/search/adapter/AreaHolder.kt
package com.constraint.vagabond.main.search.adapter

import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import com.constraint.vagabond.R
import com.squareup.picasso.Picasso

class AreaHolder internal constructor(view: View) : RecyclerView.ViewHolder(view), AreaHolderView {

    private val areaName: TextView = view.findViewById(R.id.area_name)
    private val moreInfoBtn: Button = view.findViewById(R.id.more_info)
    private val backgroundImage: ImageView = view.findViewById(R.id.card_background)

    override fun setAreaTitle(areaTitle: String) {
        areaName.text = areaTitle
    }

    override fun setAreaImage(imageUrl: String?) {
        Picasso.get()
                .load(imageUrl)
                .placeholder(R.drawable.baseline_report_problem_24)
                .fit()
                .centerCrop()
                .into(backgroundImage)
    }

    override fun setMoreInfoButton() {}
}
